x-pack/plugins/ml/public/application/explorer/annotation_timeline.tsx

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { type FC, type PropsWithChildren, useEffect } from 'react';
+import d3 from 'd3';
+import { scaleTime } from 'd3-scale';
+import { type ChartTooltipService, type TooltipData } from '../components/chart_tooltip';
+import { useCurrentThemeVars } from '../contexts/kibana';
+
+export interface AnnotationTimelineProps<T extends { timestamp: number; end_timestamp?: number }> {
+  label: string;
+  data: T[];
+  domain: {
+    min: number;
+    max: number;
+  };
+  getTooltipContent: (item: T, hasMergedAnnotations: boolean) => TooltipData;
+  tooltipService: ChartTooltipService;
+  chartWidth: number;
+}
+
+export const Y_AXIS_LABEL_WIDTH = 170;
+export const Y_AXIS_LABEL_PADDING = 8;
+const ANNOTATION_CONTAINER_HEIGHT = 12;
+const ANNOTATION_MIN_WIDTH = 8;
+
+/**
+ * Reusable component for rendering annotaiton-like items on a timeline.
```

Comment:
nit: typo in annotaiton

Reply:
fixed in https://github.com/elastic/kibana/pull/167998/commits/1f5937c65ca17b71d3dc210a9f1b360d019eebf7